### ISSUE TITLE
shape_output.txt files vary

### ISSUE THREAD

**Issue by beckerdoerthe** [awedwards/daphnia#32]
@awedwards - can you have a look at the shape_output files, please? I just checked some again, and it seems that there is something wrong. I only realized this because the size of those output files varies quite a bit. And, when looking at the data frame - some files only seem to have data for a few pictures... Any idea why this is?  Also, can you recalculate this data from the daphnia_output txt files? (or would we need to re-do our modifications again?)

**Comment by beckerdoerthe** [comment]
The weird thing is that some of the  files seem perfectly fine (those ones that are about 3M in size). But, I totally do not get why the other (those smaller ones) are missing so much data. I have only looked at a few of them now. But maybe you have a better insight.

**Comment by beckerdoerthe** [comment]
Austin, I have a guess what might be wrong. Could it be that the shape output is not 'checked-back' for whether a picture has already been done, and thus only saves the 'latest output'. I.e. if we were only to have one picture left in a pack - after saving - the shape output would only save this last data.

**Comment by beckerdoerthe** [comment]
If this is the case, I assume we need to re-do all the pic modifications. Right?

**Comment by beckerdoerthe** [comment]
Again - if this is the case - could you implement some 'skip_accepted, 1' that will include the shape output (in addition to the daphnia output) ?

**Comment by awedwards** [comment]
Oh jeez, yeah I'll check that out tomorrow. I swear I tested that, which is what took me so long to implement save_accepted in the first place... But I'll get to the bottom of it.

Can you point me to where your saved shape_output files are for your various input lists?

Thanks! Sorry about that... I will try to do my best to salvage modifications you have already made. I know that's really frustrating for you and Robert to have to do it all over again.

**Comment by beckerdoerthe** [comment]
Yeah, and I did not realise it because I must have only checked lists that I did in one go (and thus did not save in between) or files for which we tested the first few pics. And those files looked perfectly fine regarding dimensions and such.... 

So, output files are here: /mnt/spicy_4/daphnia/analysis/analysis_results
And, in case you need input files too, those are here: /mnt/pricey_3/doerthe/pheno-check/inputTXTaob

Thanks for checking on this!

**Comment by awedwards** [comment]
You are right, when truncate_output = 1 and skip_accepted = 1, only the images that are accepted in the current session are saved. I thought I tested this! Blah.

Ok, I'll try and get a fix out today. Sorry again... I really hope we are done with this mega bugs soon.

**Comment by awedwards** [comment]
Alright, I found the issue. Dumb mistake.

So I think the best way to salvage at least *some* of the fixes you've already made is to point your **output_shape_file** parameter in gui_params to your output files in the analysis_results folder. The data that was correctly saved in your output files should be fine, so you won't have to re-do those. And the ones that were skipped during save should be saved correctly now and just tack on to your output files. Does that make sense?

Sorry again!!!

**Comment by beckerdoerthe** [comment]
Yep, that makes sense. We will look into this tomorrow...

**Comment by beckerdoerthe** [comment]
Thanks!!

**Comment by beckerdoerthe** [comment]
Quick question though. Would we need to only use our previous  shape output  file (the ones in the analysis folder) or also the daphnia_output.txt as 'input' in the GUI params.txt?

Do you know what I mean?

**Comment by awedwards** [comment]
To be safe:

input_shape_file, .../shape_output.txt
output_shape_file, .../shape_output_xx_xxxxxx.txt
input_analysis_file, .../daphnia_output.txt
output_analysis_file, .../daphnia_output_xx_xxxxxx.txt

**Comment by awedwards** [comment]
So yes, use the daphnia_output.txt and shape_output.txt, from the automatic pipeline, as inputs. And use your saved files as the outputs.

**Comment by beckerdoerthe** [comment]
Ok. Got it. Thanks!

**Comment by beckerdoerthe** [comment]
Another question. Would we keep the truncate_output == 1 then? I.e., is the new code checking whether respective barcodes are in BOTH output files now ?  Just asking as our daphnia_output files will have +/- 50 data points while some of the shape_output files only have 1 data point. Does this make sense?

**Comment by awedwards** [comment]
Ah, that's a good question. Hmm... You might have to set skip_accepted == 0. You would then just quickly have to go through and manually skip the ones that look correct (i.e. accept the ones that are in the shape_output file) and re-save when you have fixed the ones that weren't saved.

**Comment by beckerdoerthe** [comment]
OK - that's what Robert suggested too. Thanks. We will check it out in the AM (and keep you posted).

**Comment by beckerdoerthe** [comment]
I just tried running the GUI and it is giving below error. @awedwards - do we need a new daphnia_output.txt and shape_output.txt (because you said you'd run the pipeline over night). If so, can you copy those files to our mnt folder?  

![screen shot 2018-12-07 at 9 18 47 am](https://user-images.githubusercontent.com/9977821/49652935-7d5da300-fa01-11e8-8613-fd7a181ef913.png)

**Comment by awedwards** [comment]
No you shouldn't need new daphnia/shape output files,just the analysis metadata file.

Could you please check your gui_params file? I had a typo in mine and got this same error.

**Comment by beckerdoerthe** [comment]
Hm, maybe I don't see it - can anybody spot any issues?

![screen shot 2018-12-07 at 10 32 51 am](https://user-images.githubusercontent.com/9977821/49656504-89e6f900-fa0b-11e8-8ddd-515a8d91185d.png)

**Comment by beckerdoerthe** [comment]
I just ran the GUI (again) - and it is actually saving a meta file now - with the pics that I modified. BUT, all the parameter are just the same as in the original meta file. So, it is not updating on the area that is masked or whether blur has been changed etc. Thoughts?

**Comment by beckerdoerthe** [comment]
BTW, everything is saved on the workstation - analysis results folder. and pic input files are in /mnt/pricey_3/doerthe/pheno-check/inputTXTaob

**Comment by awedwards** [comment]
The only thing that is changeable through the GUI are two fields: dorsal_edge_blur and flipped. The other parameters are set before running the automatic pipeline.

When you change the gaussian blur in the GUI, does it update dorsal_edge_blur? It was changing for me yesterday.

I don't know how we would store masking/unmasking data, but we can talk about that in our meeting next week.

**Comment by beckerdoerthe** [comment]
ok, that makes sense. yes, the blur is updating, actually. I think we would need to have the masked area recorded. But, not sure how much work that'd be. Let's check in on TUE on this.

**Comment by awedwards** [comment]
Sounds good. Just so I can think about it before the meeting, when you say masked area, do you want to know which edge pixels are masked? or do you want to calculate the area of the unmasked/masked area?

**Comment by beckerdoerthe** [comment]
Yes, knowing which edge pixels have been masked would be good, I think. It's more about being able to potentially reproduce the outputs we are saving now.
